importer/xmlparser/src/test/java/org/itsallcode/openfasttrace/importer/xmlparser/tree/TreeElementTest.java

```diff
@@ -0,0 +1,121 @@
+package org.itsallcode.openfasttrace.importer.xmlparser.tree;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.mockito.ArgumentMatchers.same;
+import static org.mockito.Mockito.*;
+
+import java.util.function.Consumer;
+
+import org.itsallcode.openfasttrace.api.core.Location;
+import org.itsallcode.openfasttrace.importer.xmlparser.event.StartElementEvent;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.xml.sax.Attributes;
+
+@ExtendWith(MockitoExtension.class)
+class TreeElementTest {
+    private static final String ELEMENT_URI = "uri";
+    private static final String ELEMENT_LOCAL_NAME = "localName";
+    private static final String ELEMENT_QNAME = "qname";
+
+    @Mock
+    Consumer<TreeElement> listenerMock1;
+    @Mock
+    Consumer<TreeElement> listenerMock2;
+
+    @Test
+    void testToString() {
+        assertThat(testee().toString(), equalTo(
+                "TreeElement [element=StartElementEvent [qName={uri}localName, attributeMap={qname0=Attribute [qName=qname0, value=value0]}, location=path:2], characterData=, endElementListeners=[], parent=null]"));
+    }
+
+    @Test
+    void getElement() {
+        assertThat(testee().getElement().getName().getLocalPart(), equalTo(ELEMENT_LOCAL_NAME));
+    }
+
+    @Test
+    void getCharacterDataEmpty() {
+        assertThat(testee().getCharacterData(), equalTo(""));
+    }
+
+    @Test
+    void getCharacterData() {
+        final TreeElement element = testee();
+        element.addCharacterData("chars");
+        assertThat(element.getCharacterData(), equalTo("chars"));
+    }
+
+    @Test
+    void getCharacterDataMultipleAdded() {
+        final TreeElement element = testee();
+        element.addCharacterData("chars");
+        element.addCharacterData(", more chars");
+        assertThat(element.getCharacterData(), equalTo("chars, more chars"));
+    }
+
+    @Test
+    void isRootTrueForNullParent() {
+        assertThat(testee().isRootElement(), is(true));
+    }
+
+    @Test
+    void isRootFalsForNonNullParent() {
+        final TreeElement element = new TreeElement(createStartElement(), mock(TreeElement.class));
+        assertThat(element.isRootElement(), is(false));
+    }
+
+    @Test
+    void getAttributeValueByName() {
+        assertThat(testee().getAttributeValueByName("qname0").getValue(), equalTo("value0"));
```

Comment:
This is a typical example where the fact that the test preparation is extracted, makes the test harder to validate.